app-arch/brotli/brotli-1.0.2.recipe

```diff
@@ -0,0 +1,143 @@
+libVersion="1.0.2"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+PROVIDES="
+	brotli$secondaryArchSuffix = $portVersion
+	cmd:brotli$secondaryArchSuffix = $portVersion
+	lib:libbrotlicommon$secondaryArchSuffix = $libVersionCompat
+	lib:libbrotlidec$secondaryArchSuffix = $libVersionCompat
+	lib:libbrotlienc$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	brotli${secondaryArchSuffix}_devel = $portVersion
+	devel:libbrotlicommon$secondaryArchSuffix = $libVersionCompat
+	devel:libbrotlicommon_static$secondaryArchSuffix = $libVersion
+	devel:libbrotlidec$secondaryArchSuffix = $libVersionCompat
+	devel:libbrotlidec_static$secondaryArchSuffix = $libVersion
+	devel:libbrotlienc$secondaryArchSuffix = $libVersionCompat
+	devel:libbrotlienc_static$secondaryArchSuffix = $libVersion
+	"
+REQUIRES_devel="
+	brotli$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:g++$secondaryArchSuffix
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	"
+
+PYTHON_PACKAGES=(python python3)
+PYTHON_VERSIONS=(2.7 3.6)
+# We don't have python2 for secondaryArch,
+# and gcc2 doesn't support the flags passed by python3
+if [ -n "$secondaryArchSuffix" ]; then
+	unset PYTHON_PACKAGES[0]
+	unset PYTHON_VERSIONS[0]
+elif [ "$effectiveTargetArchitecture" = "x86_gcc2" ]; then
+	unset PYTHON_PACKAGES[1]
+	unset PYTHON_VERSIONS[1]
+fi
+for i in "${!PYTHON_PACKAGES[@]}"; do
+pythonPackage=${PYTHON_PACKAGES[i]}
+pythonVersion=${PYTHON_VERSIONS[$i]}
+eval "PROVIDES_${pythonPackage}=\"\
+	${portName}_$pythonPackage = $portVersion\
+	\"; \
+REQUIRES_$pythonPackage=\"\
+	haiku\n\
+	cmd:python$pythonVersion\
+	\""
+BUILD_REQUIRES="$BUILD_REQUIRES
+	setuptools_$pythonPackage"
+BUILD_PREREQUIRES="$BUILD_PREREQUIRES
+	cmd:python$pythonVersion"
+done
+
+if [ -z "$secondaryArchSuffix" ]; then
+	REQUIRES_python+="
+		lib:libpython${PYTHON_VERSIONS[0]}
+		"
+	BUILD_PREREQUIRES+="
+		devel:libpython${PYTHON_VERSIONS[0]}
+		"
+fi
+if [ "$effectiveTargetArchitecture" != "x86_gcc2" ]; then
+	REQUIRES_python3+="
+		lib:libpython${PYTHON_VERSIONS[1]}m$secondaryArchSuffix
+		"
+	BUILD_PREREQUIRES+="
+		devel:libpython${PYTHON_VERSIONS[1]}m$secondaryArchSuffix
+		"
+fi
+
+BUILD()
+{
+	runConfigure --omit-buildspec ./configure-cmake --disable-debug
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+
+	for i in "${!PYTHON_PACKAGES[@]}"; do
+		pythonPackage=${PYTHON_PACKAGES[i]}
+		pythonVersion=${PYTHON_VERSIONS[$i]}
+
+		python=python$pythonVersion
+		installLocation=$prefix/lib/$python/vendor-packages/
+		export PYTHONPATH=$installLocation:$PYTHONPATH
+		mkdir -p $installLocation
+		rm -rf build
+		$python setup.py build install \
+			--root=/ --prefix=$prefix
+
+		packageEntries $pythonPackage \
+			$prefix/lib/python*
+	done
+
+	fixPkgconfig
+	prepareInstalledDevelLibs libbrotli{common,dec,enc}{,-static}
+	packageEntries devel "$developDir"
```

Comment:
Builds fine on x86_gcc2 primary arch. (tests fail at some point, but on x86_gcc2 this happens.)
(I haven't tested yet on x86 secondary arch, nor on x86_64.)
Would be nice to include
```
	install -d "$docDir" "$manDir"/man1 "$manDir"/man3
	install -t "$manDir"/man1 docs/brotli.1
	install -t "$manDir"/man3 docs/*.3
	install -t "$docDir" README docs/*.pdf
```
and
```diff
-	packageEntries devel "$developDir"
+	packageEntries devel \
+		"$docDir"/*.pdf \
+		"$manDir"/man3 \
+		"$developDir"
```
(Yep, it seems there is no "install-doc" Makefile target.)
Maybe it would also be nice to use the $maybe_manDir_man3 "trick" to skip the installation of the man pages on secondary arch builds.
BTW, I added quotes because I know you like them and I agree that it is better with quotes than without. This is safe in situations where the path to the haikuports tree contains special chars (e.g. spaces, tabs, etc...) and the user chooses to call "haikuporter --nochroot" if this gets fixed some day.)

Reply:
Tests fails on x86_gcc2? I would appreciate a log. Hopefully those style changes doesn't broke anything (it shouldn't).

LOL, I totally forgot about the docs :P

Reply:
Looks like only the man pages have value here. The PDF is just an comparison, and README pretty much only have instructions to build the lib